packages/opencensus-core/test/test-tracer.ts

```diff
@@ -0,0 +1,269 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {EventEmitter} from 'events';
+
+import {TracerConfig} from '../src/trace/config/types';
+import {RootSpan} from '../src/trace/model/root-span';
+import {Span} from '../src/trace/model/span';
+import {Tracer} from '../src/trace/model/tracer';
+import {OnEndSpanEventListener} from '../src/trace/model/types';
+
+class OnEndSpanClass implements OnEndSpanEventListener {
+  /** Counter for test use */
+  testCount = 0;
+  /** Happens when a span is ended */
+  onEndSpan(span: RootSpan): void {
+    this.testCount++;
+  }
+}
+
+const defaultConfig: TracerConfig = {
+  samplingRate: 1
+};
+
+describe('Tracer', () => {
+  const options = {name: 'test'};
+
+  /** Should create a Tracer instance */
+  describe('new Tracer()', () => {
+    it('should create a Tracer instance', () => {
+      const tracer = new Tracer();
+      assert.ok(tracer instanceof Tracer);
+    });
+  });
+
+  /** Should get/set the current RootSpan from tracer instance */
+  describe('get/set currentRootSpan()', () => {
+    let tracer, rootSpan;
+    before(() => {
+      tracer = new Tracer();
+      rootSpan = new RootSpan(tracer);
+      tracer.currentRootSpan = rootSpan;
+    });
+    it('should get the current RootSpan from tracer instance', () => {
+      assert.ok(tracer.currentRootSpan instanceof RootSpan);
+    });
+    it('should set the current RootSpan from tracer instance', () => {
+      assert.equal(tracer.currentRootSpan, rootSpan);
+    });
+  });
+
+  /** Should return a started tracer instance */
+  describe('start()', () => {
```

Comment:
This test seems nearly identical to `get active()`.

Reply:
TODO: Review this test case